File: ewbapp/Models/Enums/RangerRole.swift
import Foundation

enum RangerRole: String, CaseIterable, Codable {
    case ranger = "ranger"
    case seniorRanger = "seniorRanger"
    case coordinator = "coordinator"

    var displayName: String {
        switch self {
        case .ranger: return "Ranger"
        case .seniorRanger: return "Senior Ranger"
        case .coordinator: return "Coordinator"
        }
    }
}

File: app/src/main/java/org/yac/llamarangers/domain/enums/RangerRole.kt
package org.yac.llamarangers.domain.enums

enum class RangerRole(val displayName: String) {
    RANGER("Ranger"),
    SENIOR_RANGER("Senior Ranger"),
    SUPERVISOR("Supervisor");

    companion object {
        fun fromRaw(raw: String): RangerRole = when (raw) {
            "seniorRanger" -> SENIOR_RANGER
            "supervisor"   -> SUPERVISOR
            else           -> RANGER
        }
    }
}
